File: SAT.SERVICES/Services/DashboardLabService.cs
using System.Collections.Generic;
using SAT.INFRA.Interfaces;
using SAT.MODELS.Entities.Params;
using SAT.MODELS.Views;
using SAT.SERVICES.Interfaces;

namespace SAT.SERVICES.Services
{
    public class DashboardLabService : IDashboardLabService
    {
        private readonly IDashboardLabRepository _dashboardLabRepo;

        public DashboardLabService(IDashboardLabRepository dashboardLabRepo)
        {
            _dashboardLabRepo = dashboardLabRepo;
        }

        public List<ViewDashboardLabRecebidosReparados> ObterRecebidosReparados(DashboardLabParameters parameters)
        {
            return _dashboardLabRepo.ObterRecebidosReparados(parameters);
        }

        public List<ViewDashboardLabTopFaltantes> ObterTopFaltantes(DashboardLabParameters parameters)
        {
            return _dashboardLabRepo.ObterTopFaltantes(parameters);
        }

        public List<ViewDashboardLabTopTempoMedioReparo> ObterTempoMedioReparo(DashboardLabParameters parameters)
        {
            return _dashboardLabRepo.ObterTempoMedioReparo(parameters);
        }

        public List<ViewDashboardLabProdutividadeTecnica> ObterProdutividadeTecnica(DashboardLabParameters parameters)
        {
            return _dashboardLabRepo.ObterProdutividadeTecnica(parameters);
        }

        public List<ViewDashboardLabTopItensMaisAntigos> ObterTopItensMaisAntigos(DashboardLabParameters parameters)
        {
            return _dashboardLabRepo.ObterTopItensMaisAntigos(parameters);
        }

        public List<ViewDashboardLabIndiceReincidencia> ObterIndiceReincidencia(DashboardLabParameters parameters)
        {
            return _dashboardLabRepo.ObterIndiceReincidencia(parameters);
        }
    }
}

File: SAT.FRONT/src/app/core/services/dashboard-lab.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from "@angular/common/http";
import { Observable } from "rxjs";
import { map } from "rxjs/operators";
import { appConfig as c } from 'app/core/config/app.config'
import { DashboardLabParameters, ViewDashboardLabProdutividadeTecnica, ViewDashboardLabRecebidosReparados, ViewDashboardLabTopFaltantes, ViewDashboardLabTopTempoMedioReparo } from '../types/dashboard-lab.types';

@Injectable({
  providedIn: 'root'
})
export class DashboardLabService {
  constructor(private http: HttpClient) {}

  obterRecebidosReparados(parameters: DashboardLabParameters): Observable<ViewDashboardLabRecebidosReparados[]> {
    let params = new HttpParams();
    
    Object.keys(parameters).forEach(key => {
      if (parameters[key] !== undefined && parameters[key] !== null) params = params.append(key, String(parameters[key]));
    });

    return this.http.get(`${c.api}/DashboardLab/RecebidosReparados`, { params: params }).pipe(
      map((data: ViewDashboardLabRecebidosReparados[]) => data)
    )
  }

  obterTopFaltantes(parameters: DashboardLabParameters): Observable<ViewDashboardLabTopFaltantes[]> {
    let params = new HttpParams();
    
    Object.keys(parameters).forEach(key => {
      if (parameters[key] !== undefined && parameters[key] !== null) params = params.append(key, String(parameters[key]));
    });

    return this.http.get(`${c.api}/DashboardLab/TopFaltantes`, { params: params }).pipe(
      map((data: ViewDashboardLabTopFaltantes[]) => data)
    )
  }

  obterTempoMedioReparo(parameters: DashboardLabParameters): Observable<ViewDashboardLabTopTempoMedioReparo[]> {
    let params = new HttpParams();
    
    Object.keys(parameters).forEach(key => {
      if (parameters[key] !== undefined && parameters[key] !== null) params = params.append(key, String(parameters[key]));
    });

    return this.http.get(`${c.api}/DashboardLab/TempoMedioReparo`, { params: params }).pipe(
      map((data: ViewDashboardLabTopTempoMedioReparo[]) => data)
    )
  }

  obterProdutividadeTecnica(parameters: DashboardLabParameters): Observable<ViewDashboardLabProdutividadeTecnica[]> {
    let params = new HttpParams();
    
    Object.keys(parameters).forEach(key => {
      if (parameters[key] !== undefined && parameters[key] !== null) params = params.append(key, String(parameters[key]));
    });

    return this.http.get(`${c.api}/DashboardLab/ProdutividadeTecnica`, { params: params }).pipe(
      map((data: ViewDashboardLabProdutividadeTecnica[]) => data)
    )
  }
}
